std/file.d

```diff
@@ -259,15 +254,29 @@ version (Windows) void[] read(in char[] name, size_t upTo = size_t.max) @safe
     {
         return CloseHandle(hObject);
     }
-    static trustedGetFileSize(HANDLE hFile, DWORD *lpFileSizeHigh) @trusted
+    static trustedGetFileSize(HANDLE hFile, out ulong fileSize) @trusted
     {
-        return GetFileSize(hFile, lpFileSizeHigh);
+        DWORD sizeHigh;
+        DWORD sizeLow = GetFileSize(hFile, &sizeHigh);
+        bool result = sizeLow != INVALID_FILE_SIZE;
+        if (result)
+            fileSize = makeUlong(sizeLow, sizeHigh);
+        return result;
     }
-    static trustedReadFile(HANDLE hFile, void *lpBuffer, DWORD nNumberOfBytesToRead,
-                           DWORD *lpNumberOfBytesRead, OVERLAPPED *lpOverlapped) @trusted
+    static trustedReadFile(HANDLE hFile, void *lpBuffer, ulong nNumberOfBytesToRead) @trusted
     {
-        return ReadFile(hFile, lpBuffer, nNumberOfBytesToRead,
-                        lpNumberOfBytesRead, lpOverlapped);
+        // Read by chunks of size <= 4GB (Windows API limit)
+        ulong totalNumRead = 0;
+        while (totalNumRead != nNumberOfBytesToRead)
+        {
+            uint chunkSize = min(nNumberOfBytesToRead - totalNumRead, 0xffff_ffff);
```

Comment:
Reading an odd number of bytes is not too efficient. I'd use something down to some page size boundary instead of 0xFFFF_FFFF, like maybe 0xFFFF_0000.